D3 node module dependencies

#262 added D3 to the package.json - this brings in contextify (via JSDOM) as a dependency. Building this with node-gyp requires Python, Make and a C/C++ compiler on unix, and Visual Studio C++ on Windows.

Installing contextify with npm fails for me on OS X 10.10.2 with Python and the XCode command line tools installed:

> node-gyp rebuild
make: *** No rule to make target `../../../../../../../../../Volumes/Macintosh', needed by `Makefile'.  Stop.
gyp ERR! build error
...
gyp ERR! not ok 

It seems we're only doing this to have d3.js in the visual tests. Could we instead get it from http://d3js.org and download if needing it offline?

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -130,7 +130,10 @@ module.exports = function (grunt) {
         jasmine: {
             options: {
                 specs: '<%= meta.testJsFiles %>',
-                vendor: 'node_modules/d3/d3.js'
+                vendor: [
+                    'node_modules/d3/d3.js',
+                    'http://d3js.org/d3.v3.js'
+                ]
             },
             test: {
                 src: ['<%= meta.componentsJsFiles %>'],

--- a/package.json
+++ b/package.json
@@ -18,10 +18,12 @@
     "grunt-jscs": "^1.0.0",
     "grunt-jsdoc": "^0.5.7",
     "matchdep": "^0.3.0",
-    "d3": "^3.5.3",
     "css-layout": "git+https://git@github.com/ColinEberhardt/css-layout.git"
   },
   "scripts": {
     "test": "grunt ci"
+  },
+  "optionalDependencies": {
+    "d3": "^3.5.3"
   }
 }

--- a/visual-tests/index.html
+++ b/visual-tests/index.html
@@ -209,6 +209,7 @@
     </section>
 
     <script src="../node_modules/d3/d3.js"></script>
+    <script src="http://d3js.org/d3.v3.js"></script>
     <script src="../node_modules/css-layout/src/Layout.js"></script>
     <script src="../dist/d3-financial-components.js"></script>
 

